service/src/main/resources/db/migration/V244__payment.sql

```diff
@@ -0,0 +1,44 @@
+CREATE TYPE payment_status AS ENUM ('DRAFT', 'SENT');
+
+CREATE TABLE payment (
+    id uuid PRIMARY KEY NOT NULL DEFAULT ext.uuid_generate_v1mc(),
+    created timestamp with time zone NOT NULL DEFAULT now(),
+    updated timestamp with time zone NOT NULL DEFAULT now(),
+    unit_id uuid NOT NULL,
+    unit_name text NOT NULL,
+    unit_business_id text,
+    unit_iban text,
+    unit_provider_id text,
+    period daterange NOT NULL,
+    number bigint,
+    amount int NOT NULL,
+    status payment_status NOT NULL,
+    payment_date date,
+    due_date date,
+    sent_at timestamp with time zone,
+    sent_by uuid,
+
+    CONSTRAINT fk$unit FOREIGN KEY (unit_id) REFERENCES daycare (id),
+    CONSTRAINT unique$number UNIQUE (number),
+    CONSTRAINT fk$sent_by FOREIGN KEY (sent_by) REFERENCES evaka_user (id),
+    CONSTRAINT payment_state CHECK (
+        status = 'DRAFT' OR (
+            unit_business_id IS NOT NULL AND unit_business_id <> '' AND
+            unit_iban IS NOT NULL AND unit_iban <> '' AND
+            unit_provider_id IS NOT NULL AND unit_provider_id <> '' AND
+            number IS NOT NULL AND
+            payment_date IS NOT NULL AND
+            due_date IS NOT NULL AND
+            sent_at IS NOT NULL AND
+            sent_by IS NOT NULL
+        )
+    )
+);
+
+CREATE TRIGGER set_timestamp BEFORE UPDATE ON payment FOR EACH ROW EXECUTE PROCEDURE trigger_refresh_updated();
+CREATE INDEX idx$created ON payment (created);
```

Comment:
`amount` is sometimes used for `order by`, so I kept that index.

`number::text` being used with a wildcard prefix was a bug. I now changed it to to use a wildcard postfix (`like 'something%'`). Even still, I also removed the index, because it wasn't very useful.

I actually think that the SQL query text in `searchPayments` should be generated based on the filtering params instead of using the `:foo IS NULL OR (<some condition with :foo>)` pattern. Otherwise Postgres cannot plan index use correctly because it doesn't know which bind parameter is going to be `NULL` 🤔. This can be done later if the queries starts to be slow, but not that much data is actually expected in this table.